soroban-sdk/src/token.rs

```diff
@@ -0,0 +1,94 @@
+//! Token contains types for calling and accessing token contracts, including
+//! the Stellar Asset Contract.
+//!
+//! See [`Interface`] for the interface of token contracts such as the Stellar
+//! Asset Contract.
+//!
+//! See [`Client`] for calling token contracts such as the Stellar Asset
+//! Contract.
+use crate as soroban_sdk;
+use crate::{contractclient, Address, Bytes, Env};
+
+
+// The interface below was copied from
+// https://github.com/stellar/rs-soroban-env/blob/main/soroban-env-host/src/native_contract/token/contract.rs
+// at commit b3c188f48dec51a956c1380fb6fe92201a3f716b.
+//
+// Differences between this interface and the built-in contract
+// 1. The return values here don't return Results.
+// 2. The implementations have been replaced with a panic.
+// 3. &Host type usage are replaced with Env
+
+/// Interface for Token contracts, such as the Stellar Asset Contract.
+#[contractclient(name = "Client")]
+pub trait Interface {
```

Comment:
@sisuresh Instead of providing a stub contractimpl, we provide an interface that folks can `impl` if they want, as well as generate a `Client` they can use to call token contracts such as the Stellar Asset Contract.

There is one thing taht this fails to generate, and that's the functions that provide the contract spec for the token. I'm going to push some changes that'll re-add that in another fashion.

The main reason not to use `contractimpl` is that we don't really want to provide an empty impl.